### PR TITLE
Return address, tx_hash from Account.deploy_contract

### LINKED ISSUE #307: `Account.deploy_contract` should return with tx hash and/or address
`Account.deploy_contract` calls https://github.com/OpenZeppelin/nile/blob/98ce378842eb5d404f82e3bc2d082dd06091b665/src/nile/core/account.py#L198
but omits the return value.  It would be useful for the caller if this returns information such as tx hash and/or address.

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #307

### CODE REVIEW

**Review by andrew-fleming — COMMENTED** [review]
Very nice, @ericnordelo! We should probably update the return value in the mocked `deploy_contract` as well:

https://github.com/OpenZeppelin/nile/blob/main/tests/commands/test_account.py#L152

Otherwise, this looks good to me!

**Review by ericglau — APPROVED** [review]

**Review by martriay — APPROVED** [review]

**Comment by martriay** [comment]
looks good! addressing Andrew's comment should be enough

**Review by andrew-fleming — DISMISSED** [review]
After a second look, I don't think the account tests need to be updated because the deploy mocks returning values only include the other deploys (`sign_deploy`, `deploy`, and `deploy_account`) and not `deploy_with_deployer` in test_account.py.

test_deploy.py and test_declare.py are updated, so this looks good to go for me!

**Comment by martriay** [comment]
but [`deploy account`](https://github.com/OpenZeppelin/nile/pull/311/files#diff-c19a6bc8b864ba31b67d13c034aa2eba8da5047cfa2683bb708ce6773fbfdf21R187) did change its return value

File: src/nile/core/account.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ async def deploy_contract(
             deployer_address or UNIVERSAL_DEPLOYER_ADDRESS
         )
 
-        await deploy_with_deployer(
+        return await deploy_with_deployer(
```
Comment (martriay) [inline src/nile/core/account.py:198]: this should be tested too

File: tests/commands/test_account.py [hunk 2]
```diff
@@ -320,7 +349,10 @@ async def test_send_defaults(
 
 
 @pytest.mark.asyncio
-@patch("nile.core.account.deploy_account", return_value=(MOCK_ADDRESS, MOCK_INDEX))
+@patch(
+    "nile.core.account.deploy_account",
+    return_value=(MOCK_ADDRESS, MOCK_TX_HASH, MOCK_ABI),
```
Comment (martriay) [inline tests/commands/test_account.py:354]: it looks like we could abstract the whole tuple into a variable, to simplify these changes in the future

**Review by martriay — APPROVED** [review]

File: tests/commands/test_account.py (hunk 2 above)
Comment (ericnordelo) [reply tests/commands/test_account.py]: Done!

**Review by martriay — APPROVED** [review]

**Pull request merged**